scripts/irods/test/test_resource_types.py

```diff
@@ -2385,6 +2385,153 @@ def test_auto_repl_on__2941(self):
         self.admin.assert_icommand("ils -L " + filename, 'STDOUT_SINGLELINE', 'cacheResc')
         self.admin.assert_icommand("ils -L " + filename, 'STDOUT_SINGLELINE', 'archiveResc')
 
+    def test_get_stale_cache_replica(self):
+
+        try:
+            filename = "testgetstalecachereplica.txt"
+            lib.make_file(filename, 10, 'arbitrary')
+            stale_replica_filename = "stalereplica.txt"
+            lib.make_file(stale_replica_filename, 20, 'arbitrary')
+
+            logical_path = os.path.join(self.admin.session_collection, filename)
+
+            self.admin.assert_icommand("iput " + filename)
+            self.admin.assert_icommand("iadmin modresc demoResc context \"auto_repl=off\"" )
+            self.admin.assert_icommand("iput -f " + stale_replica_filename + " " + filename)
+
+            # at this point the cache (repl 0) will be good and the archive (repl 1) will be stale
+            # swap the status of these two so archive is good and cache is stale
+            self.admin.assert_icommand("iadmin modrepl logical_path " + logical_path + " replica_number 0 DATA_REPL_STATUS 0")
+            self.admin.assert_icommand("iadmin modrepl logical_path " + logical_path + " replica_number 1 DATA_REPL_STATUS 1")
+
+            # get the stale replica
+            get_filename = "getfile.txt"
+            self.admin.assert_icommand("iget -n 0 " + filename + " " +  get_filename)
+
+            # assert on the stale replica file size
+            self.assertEqual(os.path.getsize(get_filename), 20)
+
+            # get the file without selecting the replica
+            self.admin.assert_icommand("iget -f " + filename + " " +  get_filename)
+
+            # assert on the good replica file size
+            self.assertEqual(os.path.getsize(get_filename), 10)
+
+        finally:
+            # cleanup
+            self.admin.assert_icommand("irm -f " + filename)
+            if os.path.exists(filename):
+                os.remove(filename)
+            if os.path.exists(stale_replica_filename):
+                os.remove(stale_replica_filename)
+            if os.path.exists(get_filename):
+                os.remove(get_filename)
```

Comment:
Didn't you add something to lib.py recently that makes this easy?

I thought we introduced something like `lib.remove_if_exists()` somewhere. It may have been in the S3 resouce plugin.

Consider adding that to lib.py and using it for these.

Reply:
I didn't yet but there is an issue on it.  Will put it in this pull request so that I can make the changes above.

Reply:
done